src/test/scala/nl.knaw.dans.easy.deposit/docs/DatasetXmlSpec.scala

```diff
@@ -22,22 +22,73 @@ import better.files.File
 import javax.xml.XMLConstants
 import javax.xml.transform.Source
 import javax.xml.transform.stream.StreamSource
-import javax.xml.validation.SchemaFactory
+import javax.xml.validation.{ Schema, SchemaFactory }
 import nl.knaw.dans.easy.deposit.TestSupportFixture
 import nl.knaw.dans.easy.deposit.docs.DatasetMetadata.{ DateQualifier, QualifiedSchemedValue }
 import resource.Using
 
 import scala.util.{ Failure, Success, Try }
-import scala.xml.{ Node, Utility }
+import scala.xml.{ Elem, Node, PrettyPrinter, Utility }
 
-class DatasetXmlSpec extends TestSupportFixture {
+trait DdmBehavior {
+  this: TestSupportFixture =>
 
-  private lazy val triedSchema = Try { // loading postponed until we actually start validating
+  lazy val triedSchema: Try[Schema] = Try { // loading postponed until we actually start validating
     val factory = SchemaFactory.newInstance(XMLConstants.W3C_XML_SCHEMA_NS_URI)
     val schemas = "http://dublincore.org/schemas/xmls/qdc/2008/02/11/dcterms.xsd" ::
       "https://easy.dans.knaw.nl/schemas/md/ddm/ddm.xsd" :: Nil
     factory.newSchema(schemas.map(xsd => new StreamSource(xsd)).toArray[Source])
   }
+  val emptyDDM: Elem
+
+  // pretty provides friendlier trouble shooting for complex XML's than Utility.trim
+  private val prettyPrinter: PrettyPrinter = new scala.xml.PrettyPrinter(1024, 2)
+
+  /**
+   * @param input          a Success(DatasetMetadata)
+   * @param expectedOutput <ddm:profile> and optionaly <ddm:dcmiMetadata>
+   *                       if provided, input should convert into a DDM with the same content
+   */
+  def validDatasetMetadata(input: Try[DatasetMetadata], expectedOutput: Seq[Node]): Unit = {
+    lazy val datasetMetadata = input.recoverWith { case e =>
```

Comment:
`recoverWith` just to do a side-effect is called `doIfFailure` (see dans-scala-lib)